Implemented edit request feature to allow users to modify their own r…

okikiola-jkt opened the pull request:
…equests, including proper authorization checks and database updates

okikiola-jkt committed:
Implemented edit request feature to allow users to modify their own requests, including proper authorization checks and database updates

fob413 commented on src/controllers/requests/editRequests.js:
```diff
@@ -0,0 +1,44 @@
+require('dotenv').config();
+
+const getRequestsById = async (id) => {
+    const query = 'SELECT * from requests WHERE id = $1';
```
You should query with both id and userId, so that you're sure that the user trying to edit is the owner

fob413 commented on src/controllers/requests/editRequests.js:
```diff
@@ -0,0 +1,44 @@
+
+const editRequest = async (request, response) => {
+    try{
+        const {details, id} = request.body;
```
the id should be gotten from the url, not from the request body

fob413 commented on src/controllers/requests/editRequests.js:
```diff
@@ -0,0 +1,44 @@
+            })
+        }
+
+        if (existingRequest.userid !== request.user.id) {
```
this can be accomplished from the query 😁 from one of the comment above

fob413 commented on src/controllers/requests/editRequests.js:
```diff
@@ -0,0 +1,44 @@
+        await db.query(updateQuery, [details, id]);
+
+
+        return response .status(201).json({
```
The status code should be 200. 201 is for when something new is being created, but in this scenario, we're not creating something new, we're updating something that already exists

fob413 commented on src/controllers/requests/editRequests.js:
```diff
@@ -0,0 +1,44 @@
+        })
+    } catch (error) {
+        return response.status(500).json({
+            status: "Failed",
```
The status `Failed` here might cause an error on the frontend, because we've been using `status: 'error'`. So we should try to keep the standard everywhere, so that on the frontend, we have an idea about what's we're expecting from the backend

fob413 commented on src/controllers/requests/newRequests.js:
```diff
@@ -28,8 +28,8 @@ const addNewRequest = async (request, response) => {
     } catch (error) {
         return response.status(500).json({
             status: "Failed",
```
Same comment as above. Let's keep the status standard throughout the backend.

fob413 commented on src/middleware/api/auth.js:
```diff
@@ -9,7 +9,7 @@ const getUserById = async (id) => {
 const apiAuth = async (request, response, next) => {
     try{
-        const token = request.headers.authorization.split(' ')[1];
+        const token = request.headers.authorization;
```
Is there a reason you changed this back?

okikiola-jkt replied:
The bearer was not working initially and I thought we changed it back when we were reviewing the code.

fob413 reviewed (changes requested):
Dropped a couple of comments here

fob413 commented on src/routes.js:
```diff
@@ -29,4 +30,6 @@ app.post(
     addNewRequest
 );
 
+app.put('/request/:id', apiAuth, editRequest);
```
Add in the middleware to validate the request body. You can use the same with one you used for the create request

okikiola-jkt committed:
Implemented changes based on FOB feedback
updated edituser.js to get url parameter
Implemented middlware for editing requests

fob413 commented on src/middleware/requests/editRequestMiddleware.js:
```diff
@@ -0,0 +1,18 @@
+const Joi = require ('joi')
```
You don't need to create this new middleware, because its exactly the same as `newRequestsMiddleware.js`. You can just reuse `newRequestsMiddleware.js` instead of creating a new one that is exactly the same, with no difference

okikiola-jkt committed:
made changes to middleware used for editing request

fob413 reviewed (approved):
LGTM
